modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/AbstractFilterToMongo.java

```diff
@@ -107,6 +108,8 @@ public abstract class AbstractFilterToMongo implements FilterVisitor, Expression
     /** The whole world in WGS84 */
     private static final Envelope WORLD = new Envelope(-179.99, 179.99, -89.99, 89.99);
 
+    private static final Set<Class<?>> aggregateClasses = Set.of(DWithin.class);
```

Comment:
I'm confused here... in SQL land an aggregate computes something, like min/max/count and so on.
[The MongoDB documentation seems to be suggesting the same.](https://www.mongodb.com/docs/manual/aggregation/).

Is "aggregate" the right name to use here? Please elaborate.

Reply:
Now mongo delegates DWithin operation to $geoNear. And [$geoNear](https://www.mongodb.com/docs/manual/reference/operator/aggregation/geoNear/#mongodb-pipeline-pipe.-geoNear) is a aggregation pipeline step, it can't be used in ordinary query like "find", and instead we should use "aggregate" and provide list of such steps. I should probably rename it to something like "operationsUsedByAggregation" or "aggregationSteps" and add comment to explain that further. At least it's how I understand it (don't know mongo that well)

Reply:
Removed this